omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/BaseResource1_12Test.java

```diff
@@ -0,0 +1,145 @@
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_12;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Assert;
+import org.junit.Before;
+import org.openmrs.Concept;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.HandlerExecutionChain;
+import org.springframework.web.servlet.mvc.annotation.AnnotationMethodHandlerAdapter;
+import org.springframework.web.servlet.mvc.annotation.DefaultAnnotationHandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+import static org.hamcrest.core.Is.is;
+
+public abstract class BaseResource1_12Test extends BaseModuleWebContextSensitiveTest {
```

Comment:
I see. I'm not familiar with the Allergy API and UI thus cannot make any assessments. I've just reviewed your changes from the webservices.rest module perspective and how we typically implement resources. I'll let others who worked on Allergy API to chime in.